File: backend/models/questions.go
package models

import (
	"github.com/lib/pq"
	"gorm.io/gorm"
)

// Questions構造体は、クイズの質問を表現する
type Questions struct {
	gorm.Model                    // IDフィールドを含む構造体
	UserQuestionID *string        `gorm:"type:text"` // ユーザー管理用の質問ID, NULLを許容 Goの仕様上stringはnilを許容しないので*とする
	Question       string         `gorm:"type:text;not null"`
	Options        pq.StringArray `gorm:"type:text[];not null"` // Options[0]を正解とする
	Supplement     string         `gorm:"type:text;not null"`
	Difficulty     uint           `gorm:"type:integer;not null"`
}

File: frontend/src/data/Questions.ts
// Questions.ts

import { Question } from '../types';
  
  export const questions: Question[] = [
    {
      id: 1, // GORM.Modelから
      question: "サンプル質問1",
      options: ["選択肢1", "選択肢2", "選択肢3", "選択肢4"],
      supplement: "補足情報1",
      difficulty: 1,
    },
    {
      id: 2, // 
      question: "サンプル質問2",
      options: ["選択肢A", "選択肢B", "選択肢C", "選択肢D"],
      supplement: "補足情報2",
      difficulty: 2,
    },
  ];
  